data/mods/megasforall/conditions.ts

```diff
@@ -196,4 +196,319 @@ export const Conditions: {[k: string]: ConditionData} = {
+			const move = this.dex.getMove(data.move);
+			this.add('-ability', this.effectData.source, 'Long Whip');
+			if (!data.target) {
+				this.hint(`${move.name} did not hit because there was no target.`);
+				return;
+			}
+
+			this.add(`${data.target.name} took the ${move.name} attack!`);
+			data.target.removeVolatile('Protect');
+			data.target.removeVolatile('Endure');
+
+			if (data.source.hasAbility('infiltrator') && this.gen >= 6) {
+				data.moveData.infiltrates = true;
+			}
+			if (data.source.hasAbility('normalize') && this.gen >= 6) {
+				data.moveData.type = 'Normal';
+			}
+			if (data.source.hasAbility('adaptability') && this.gen >= 6) {
+				data.moveData.stab = 2;
+			}
+			data.moveData.isFutureMove = true;
+			data.move.multihit = null;
+
+			this.useMove(move, data.source, data.target);
+		},
+	},
+	longwhip2: {
+		// this is a slot condition
+		name: 'longwhip2',
+		onResidualOrder: 3,
+		onResidual(target) {
+			// unlike a future move, Long Whip activates each turn
+			this.effectData.target = this.effectData.side.active[this.effectData.position];
+			const data = this.effectData;
+			const move = this.dex.getMove(data.move);
+			this.add('-ability', this.effectData.source, 'Long Whip');
+			if (!data.target) {
+				this.hint(`${move.name} did not hit because there was no target.`);
+				return;
+			}
+
+			this.add(`${data.target.name} took the ${move.name} attack!`);
+			data.target.removeVolatile('Protect');
+			data.target.removeVolatile('Endure');
+
+			if (data.source.hasAbility('infiltrator') && this.gen >= 6) {
+				data.moveData.infiltrates = true;
+			}
+			if (data.source.hasAbility('normalize') && this.gen >= 6) {
+				data.moveData.type = 'Normal';
+			}
+			if (data.source.hasAbility('adaptability') && this.gen >= 6) {
+				data.moveData.stab = 2;
+			}
+			data.moveData.isFutureMove = true;
+			data.move.multihit = null;
+
+			this.useMove(move, data.source, data.target);
+		},
+		onEnd(target) {
+			// unlike a future move, Long Whip activates each turn
+			this.effectData.target = this.effectData.side.active[this.effectData.position];
+			const data = this.effectData;
+			const move = this.dex.getMove(data.move);
+			this.add('-ability', this.effectData.source, 'Long Whip');
+			if (!data.target) {
+				this.hint(`${move.name} did not hit because there was no target.`);
+				return;
+			}
+
+			this.add(`${data.target.name} took the ${move.name} attack!`);
+			data.target.removeVolatile('Protect');
+			data.target.removeVolatile('Endure');
+
+			if (data.source.hasAbility('infiltrator') && this.gen >= 6) {
+				data.moveData.infiltrates = true;
+			}
+			if (data.source.hasAbility('normalize') && this.gen >= 6) {
+				data.moveData.type = 'Normal';
+			}
+			if (data.source.hasAbility('adaptability') && this.gen >= 6) {
+				data.moveData.stab = 2;
+			}
+			data.moveData.isFutureMove = true;
+			data.move.multihit = null;
+
+			this.useMove(move, data.source, data.target);
+		},
+	},
+	longwhip3: {
```

Comment:
What's the difference between the five `longwhip`s? Is there one?

Reply:
There is not a difference! It would actually be _very_ helpful to combine them into one if there is a way, especially since it sounds like the mod might be changing how it works in the future
The reason I have five of them right now is because it's supposed to be possible to stack five of them at once, but they have discrete durations (so, like, longwhip1 can last five turns, and longwhip2 can last five turns, and those five turns can overlap but not start and end at the same time), and I wasn't sure how to combine that into one slot condition
(I'm sure it's obvious but I am pretty new at this OTL)
But yeah, if there don't need to be five of them for that to work, I would massively appreciate any advice you have there

Reply:
I'm not familiar with simulation code but it seems like you could just run `longwhip` five times? If that doesn't work, put a `const longwhip = {...};` at the top of the file, and then just set all the longwhips to that. 
```ts
longwhip1: longwhip,
longwhip2: longwhip,
longwhip3: longwhip,
```

Reply:
Oh, that's actually so cool! I never knew you could do that, but it saves so much space! That will be a lot less of a headache to revise if it's supposed to work differently again in the future. XP
Thank you so much for the advice!! (and thank you Kris for correcting how I did it at first!)